src/concurrency/weak_memory.rs

```diff
@@ -117,6 +117,26 @@ impl StoreBufferAlloc {
         Self { store_buffers: RefCell::new(AllocationMap::new()) }
     }
 
+    /// When a non-atomic access happens on a location that has been atomically accessed
+    /// before without data race, we can determine that the non-atomic access fully happens
+    /// before all the prior atomic accesses so the location no longer needs to exhibit
+    /// any weak memory behaviours until further atomic accesses.
```

Comment:
"fully happens before all prior atomic accesses" sounds wrong, I assume you mean "after"?